src/compress.rs

```diff
@@ -146,11 +148,40 @@ mod snappy {
 		}
 
 		pub(super) fn compress(&self, value: &[u8]) -> Vec<u8> {
-			snappy::compress(value)
+			let mut buf = Vec::with_capacity(value.len() << 3);
```

Comment:
Did initializing without capacity results in worst benchmark?
(in a branch I had snap slower than snappy on macro testing (compress a whole snapshot),)

Reply:
Nevermind there was something fishy in my branch.

Reply:
I still had my test snapshot, reran it, no significant difference(snap or snappy, with or without using with_capacity all took me around 55 secs to compress a full polkadot snapshot).
@gilescope if you are interested it is in https://github.com/cheme/parity-db/tree/compression , there is also other tests.
(got some command interface and admin iteration on db content in this branch and probably also in command branch).